core/domain/user_services.py

```diff
@@ -656,3 +656,36 @@ def get_user_dashboard_stats(user_id):
             'total_plays': 0,
             'average_ratings': None
         }
+
+def get_weekly_dashboard_stats(user_id):
+    """Returns list which contains the dashboard stats of a user"""
+
+    model = user_models.UserStatsModel.get(user_id, strict=False)
+
+    if model and model.weekly_creator_stats_list:
+        return model.weekly_creator_stats_list
+    else:
+        return None
+
+def save_last_dashboard_stats(user_id):
+    """Save statistics for creator dashboard of a user by appending to a list
+    keyed by datetime.utcnow().date().
+
+    This method is called through a cron job which is run weekly.
+    """
+    model = user_models.UserStatsModel.get(user_id, strict=False)
+    if not model:
```

Comment:
Actually, I think it makes sense to do it this way, since our continuous job updating UserStatsModel creates new models for users. 

As a side note, the UserStats MR job used to only create models for users with non-zero plays/ratings. Do you think it will be a good idea to modify the current UserStats job to do the same? This way we can avoid storing a lot of unnecessary data.